content/calloop-v0.7.markdown

```diff
@@ -0,0 +1,121 @@
+async/await if you never need to `.await` anything? This is for a large part our context of
+work in Smithay, and so we developped calloop.
+
+Calloop is designed trying to ease the friction with Rust as much as possible. The central
+issue, state sharing, is managed though some global shared data. When you're running your event
+loop, you can provide the method with a `&mut Data` reference, that is then passed down to all
+of your callbacks. If this `Data` contains the state of your app, all your callbacks can then
+access it as if they were all methods of the same object.
+
+In a nutshell, calloop is mostly targetted at apps that spend a large portion of their time just
+waiting for something to happen, and need to quickly react to it. As opposed to apps that need
+to manage numerous computations or sockets in parallel efficiently, which is the more common
+use of async/await. This processing can be suited to GUI apps notably, in particular on X11 or
+Wayland contexts, where the communication with the display server is made over an Unix socket,
+which can be monitored by APIs like `epoll` or `kqueue`.
+
+## How is it used?
+
+Calloop's core is the `EventLoop` type. As you can tell by its name, it is the event loop. Once
+you've prepared everything, you just call `EventLoop::run(...)` and that method will block, running
+your event loop and invoking the appropriate callbacks at the appropriate moment. Setting up these
+callbacks is done though two main components: the `LoopHandle` object, and the `EventSource` trait.
+
+The `LoopHandle` is a handle to the `EventLoop` (duh). This handle can be cloned and shared in
+different parts of your program. This handle is what is used to setup new callbacks, which can thus
+be done from mostly anywhere, even within a callback! The notable constraint being that the handle is
+not shareable between threads. Calloop is meant to run the event loop on a single thread.
+
+The `EventSource` trait is a generic interface for "something that can generate events". Anything
+implementing that trait can be associated to a callback in the `EventLoop`. It will be monitored
+and the associated callback will be invoked every time the event source generates an event. This
+trait is meant to be composable: one can implement a high-level event source by wrapping a low-level
+one and further processing its event before passing them along.
+
+To give you a feel of what it looks like, take a look at this example extracted from the documentation:
+
+```rust
+extern crate calloop;
+
+use calloop::{generic::Generic, EventLoop, Interest, Mode};
+
+use std::time::Duration;
+
+fn main() {
+    // Create the event loop
+    let mut event_loop = EventLoop::try_new()
+                .expect("Failed to initialize the event loop!");
+    // Retrieve an handle. It is used to insert new sources into the event loop
+    // It can be cloned, allowing you to insert sources from within source callbacks
```

Comment:
```suggestion
    // It can be cloned, allowing you to insert sources from within callbacks
```

or is there some other sort of callback in this context?

Reply:
If there is, I am confused ^^